Update syntax docs (#5615)
nextflow-io/nextflow: docs/vscode.md

}
```

**Spread operator**

> Groovy supports a "spread" operator, which can be used to flatten a list in certain situations:

Reviewer: ```suggestion
In Groovy, the "spread" operator can be used to flatten a list:
```

Reviewer: Does this only apply in certain situations? Because the certain situations aren't addressed explicitly I've removed it, but if you think it's an important point we can add it back/

```groovy
ch.map { meta, bambai -> [meta, *bambai] }
```

> The Nextflow language specification does not support the spread operator. Enumerate the list elements explicitly instead, for example:

Reviewer: ```suggestion
The Nextflow language specification does not support the spread operator. Enumerate the list elements explicitly instead:
```

To ease the migration of existing scripts, the language server only reports warnings for Groovy-style type annotations and implicit variable declarations. These warnings will become errors in the future.

:::{note}

> Because type annotations are useful in practice for providing type checking at runtime, the language server will not report errors or warnings for Groovy-style type annotations, so as to not burden existing scripts that make heavy use of them.
> 
> Type annotations will be addressed in a future version of the Nextflow language specification, at which point the language server will provide a way to automatically migrate Groovy-style type annotations to the new syntax.

Reviewer: ```suggestion
Type annotations are useful for providing type checking at runtime. To avoid burdening existing scripts that use them heavily, the language server will not report errors or warnings for Groovy-style type annotations.

Type annotations will be addressed in a future version of the Nextflow language specification. At this point, the language server will provide a way to automatically migrate Groovy-style type annotations to the new syntax.
```

"""
```

**Type conversions**

Groovy supports two ways to perform type conversions:

```groovy
def map = (Map) readJson(json)  // soft cast
def map = readJson(json) as Map // hard cast
```

> The Nextflow language specification only supports hard casts. However, hard casts are discouraged because they can cause unexpected behavior if used improperly. You can use a Groovy-style type annotation instead:

Reviewer: ```suggestion
The Nextflow language specification only supports hard casts. However, hard casts are discouraged because they can cause unexpected behavior if used improperly. Use a Groovy-style type annotation instead:
```

`nextflow.java.home`
: Specifies the folder path to the JDK. Use this setting if the extension cannot find Java automatically.

`nextflow.paranoidWarnings`

> : Enable additional warnings for things like future deprecations, discouraged patterns, and so on.

Reviewer: ```suggestion
: Enable additional warnings for future deprecations, discouraged patterns, and other upcoming changes.
```
Are these additional warnings? Or are these warnings only shown if this is enabled?

Author: They are both
